### PR TITLE
feat(cmd+deps): remove promptui dep and move inputs to charm/huh

### PR DESCRIPTION
**Pull request by rektdeckard**
- Removes dependency on [promptui](https://github.com/manifoldco/promptui), which was EOL
- Moves interactive prompts to [charmbracelet/huh](https://github.com/charmbracelet/huh), not to be confused with @cacheonly 

https://github.com/livekit/livekit-cli/assets/26732044/1317d375-f832-4983-8010-40fd839d6c3e

### CODE REVIEW

File: cmd/lk/token.go [hunk 1]
```diff
@@ -19,14 +19,27 @@ import (
...
 
+	"github.com/charmbracelet/huh"
 	"github.com/urfave/cli/v3"
 
 	"github.com/livekit/protocol/auth"
 	"github.com/livekit/protocol/livekit"
 )
 
+const (
+	usageCreate   = "Token bearer can create or delete rooms"
```
Comment (davidzhao) [inline cmd/lk/token.go:33]: seems a bit repetitive, what about just dropping the `Token bearer can` or replace with `ability to`

```suggestion
	usageCreate   = "Token bearer can create or delete rooms"
```

File: cmd/lk/token.go [hunk 2]
```diff
@@ -266,7 +279,61 @@ func createToken(ctx context.Context, c *cli.Command) error {
...
+
+		permissions := make([]permission, 0)
+
+		if err := huh.NewMultiSelect[permission]().
+			Options(
+				huh.NewOption("Create", pCreate),
+				huh.NewOption("List", pList),
+				huh.NewOption("Join", pJoin),
+				huh.NewOption("Admin", pAdmin),
+				huh.NewOption("Record", pRecord),
```
Comment (davidzhao) [inline cmd/lk/token.go:303]: we don't want to issue `Record` permissions. for context, Egress instance will create their own tokens and join the room setting the `recorder` token and `participant.kind=egress`

**Review by davidzhao — COMMENTED** [review]
nice! just a couple of nits

File: cmd/lk/project.go [hunk 3]
```diff
@@ -123,106 +154,85 @@ func addProject(ctx context.Context, cmd *cli.Command) error {
...
-		prompt = promptui.Prompt{
-			Label:    "API Key",
-			Validate: validateKey,
-		}
-		if p.APIKey, err = prompt.Run(); err != nil {
-			return err
-		}
+		prompts = append(prompts, huh.NewInput().
+			Title("API Key").
+			Placeholder("APIGwek9pfUvb2s").
```
[the remark this answers is not in the document]
Comment (rektdeckard) [reply cmd/lk/project.go:181]: Good call. I saw that from in the docs too.

File: cmd/lk/project.go [hunk 4]
```diff
@@ -123,106 +154,85 @@ func addProject(ctx context.Context, cmd *cli.Command) error {
...
-		// cannot conflict with existing projects
-		for _, p := range cliConfig.Projects {
-			if p.Name == val {
-				return errors.New("name already exists")
-			}
-		}
-		return nil
+		prompts = append(prompts, huh.NewInput().
+			Title("API Secret").
+			Placeholder("B7CZPLqWNoNMnXGWuJeFoBYIvBdPkAnyOGMzpcrfzApn").
```
[the remark this answers is not in the document]
Comment (rektdeckard) [reply cmd/lk/project.go:195]: XXxXxxxXxXxxxXxXXxXXx?

[the remark this answers is not in the document]
Comment (davidzhao) [reply cmd/lk/project.go:195]: or `******` ?

File: cmd/lk/project.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (davidzhao) [reply cmd/lk/project.go:181]: docs is actually giving you a valid one for your project :)

File: cmd/lk/token.go (hunk 2 above)
Comment (rektdeckard) [reply cmd/lk/token.go:303]: Ah ok. So remove this flag altogether?

Comment (davidzhao) [reply cmd/lk/token.go:303]: yeah. that'd be great!

**Review by davidzhao — APPROVED** [review]

**Pull request merged**